android/src/main/java/com/chargebee/android/reactnative/utils/Converter.kt

```diff
@@ -30,22 +35,56 @@ internal fun convertReadableArray(readableArray: ReadableArray): ArrayList<Strin
 internal fun convertListToWritableArray(array: List<CBProduct>): WritableArray {
   val writableArray: WritableArray = WritableNativeArray()
   for (item in array) {
-    writableArray.pushMap(convertProductToDictionary(item))
+    val productMaps = item.toDictionary()
+    productMaps.forEach{ writableArray.pushMap(it) }
   }
   return writableArray
 }
 
-internal fun convertProductToDictionary(product: CBProduct): WritableMap {
+internal fun CBProduct.toDictionary(): List<WritableMap> {
+  val maps: ArrayList<WritableMap> = ArrayList()
+  subscriptionOffers?.forEach{
+    maps.add(it.toMap(this))
+  }
+  oneTimePurchaseOffer?.let { maps.add(it.toMap(this)) }
+  return maps
+}
+
+fun SubscriptionOffer.toMap(product: CBProduct): WritableMap {
+  val pricingPhase = pricingPhases.last()
```

Comment:
@cb-haripriyan Code optimized for readability with apply if it makes sense 

return WritableNativeMap().apply {
        putString("productId", product.productId)
        putString("productTitle", product.productTitle)
        putDouble("productPrice", convertPriceAmountInMicros(product.skuDetails))
        putString("productCurrencyCode", product.skuDetails.priceCurrencyCode)
        putString("productId", product.id)
        putString("productType", product.type.value)
        putString("baseProductId", basePlanId)
        putString("offerToken", offerToken)
        putString("offerTitle", product.title)
        putDouble("offerPrice", pricingPhase.convertPriceAmountInMicros())
        putString("offerCurrencyCode", pricingPhase.currencyCode)
        
        offerId?.let { putMap("offer", offer()) }
    }